### PR TITLE
errors in json, column not found -> 404

### PR DESCRIPTION
**Pull request by sokrahta**
+ error messages formatted in json
+ column not found returns http 404

- will merge from master

[Review on Reviewable](https://reviewable.io/reviews/socrata/spandex/35)

### CODE REVIEW

File: spandex-http/src/main/scala/com/socrata/spandex/http/SpandexServlet.scala [hunk 1]
```diff
@@ -54,7 +59,14 @@ class SpandexServlet(conf: SpandexConfig,
     logger.info(s"GET /suggest $datasetId|$copyNum|$userColumnId :: $text / fuzz:$fuzz size:$size")
 
     val column: ColumnMap = client.getColumnMap(datasetId, copyNum, userColumnId)
-      .getOrElse(halt(HttpStatus.SC_BAD_REQUEST, s"column '$userColumnId' not found"))
+      .getOrElse(halt(
```
Comment (urmilan) [inline spandex-http/src/main/scala/com/socrata/spandex/http/SpandexServlet.scala]: I'd rather create an error response object with a json codec, than return strings here. It's just too easy to make a typo and return invalid or inconsistent JSON. You could also do what Geospace does and set up automatic serialization via Scalatra, and just return Maps.

**Comment by urmilan** [comment]
Hmm, looks like PR builder won't work until the merge conflict is resolved.

File: spandex-http/src/main/scala/com/socrata/spandex/http/SpandexServlet.scala (hunk 1 above)
Comment (velvia) [inline spandex-http/src/main/scala/com/socrata/spandex/http/SpandexServlet.scala]: One of the many ideas I've had is to make rojoma-json work with Scalatra.

**Comment by sokrahta** [comment]
thanks for the comment wrt error response object, will fix.
going to start a fresh pull request.

**Pull request closed without merging**